File: proto/server-old/src/log.rs
pub enum Level {
    VERBOSE,
    DEBUG,
    INFO,
    WARN,
    ERROR
}

pub fn log(level: Level, msg: &str) {
    println!("log: {}", msg)
}

pub fn verbose(msg: &str) {
    log(Level::VERBOSE, msg)
}

pub fn debug(msg: &str) {
    log(Level::DEBUG, msg)
}

pub fn info(msg: &str) {
    log(Level::INFO, msg)
}

pub fn warn(msg: &str) {
    log(Level::ERROR, msg)
}

pub fn error(msg: &str) {
    log(Level::ERROR, msg)
}

File: lib/js/src/log.js
import winston from '../../desktop-app/node_modules/winston'

export const logger = new winston.Logger({
  transports: [
    new winston.transports.Console({
      level: 'debug',
      json: false,
      timestamp: false,
      colorize: 'all',
    }),
    //new winston.transports.File({ filename: __dirname + '/debug.log', json: false })
  ],
  exitOnError: false
})

export const ifnerr = (err, errMsg, successMsg, resolve) => {
  if (err) {
    logger.error(`${errMsg} ${err}`)
  }
  else {
    if (successMsg) {
      logger.info(successMsg)
    }
    if (resolve) {
      resolve()
    }
  }
}
